src/utils/use-onboarding-get-custom-should-show.ts

```diff
@@ -0,0 +1,138 @@
+import {useGeolocationState} from '@atb/GeolocationContext';
+import {useRemoteConfig} from '@atb/RemoteConfigContext';
+
+import {useAuthState} from '@atb/auth';
+import {useShouldShowShareTravelHabitsScreen} from '@atb/beacons/use-should-show-share-travel-habits-screen';
+import {useMobileTokenContextState} from '@atb/mobile-token';
+import {
+  useNotifications,
+  usePushNotificationsEnabled,
+} from '@atb/notifications';
+import {useHasFareContractWithActivatedNotification} from '@atb/notifications/use-has-fare-contract-with-activated-notification';
+
+import {useValidRightNowFareContract} from '@atb/ticketing/use-valid-right-now-fare-contracts';
+
+import {Platform} from 'react-native';
+
+import {useCallback} from 'react';
+import {OnboardingSectionId} from './use-onboarding-sections';
+import {useOnboardingSectionIsOnboarded} from './use-onboarding-section-is-onboarded';
+
+export const useOnboardingGetCustomShouldShow = (
+  utilizeThisHookInstanceForSessionCounting: boolean,
+) => {
+  const customShouldShowExtendedOnboarding =
+    useCustomShouldShowExtendedOnboarding();
+
+  const customShouldShowLocationOnboarding =
+    useCustomShouldShowLocationOnboarding();
+
+  const customShouldShowShareTravelHabitsScreen =
+    useShouldShowShareTravelHabitsScreen(
+      utilizeThisHookInstanceForSessionCounting,
+    );
+
+  const customShouldShowNotificationPermissionScreen =
+    useCustomShouldShowNotificationPermissionScreen();
+
+  const customShouldShowMobileTokenOnboarding =
+    useCustomShouldShowMobileTokenOnboarding();
+
+  const customShouldShowMobileTokenWithoutTravelcardOnboarding =
+    useCustomShouldShowMobileTokenWithoutTravelcardOnboarding();
+
+  return useCallback(
+    (onboardingSectionId: OnboardingSectionId) => {
+      switch (onboardingSectionId) {
+        case 'extendedOnboarding':
+          return customShouldShowExtendedOnboarding;
+        case 'locationWhenInUsePermission':
+          return customShouldShowLocationOnboarding;
+        case 'shareTravelHabits':
+          return customShouldShowShareTravelHabitsScreen;
+        case 'notificationPermission':
+          return customShouldShowNotificationPermissionScreen;
+        case 'mobileToken':
+          return customShouldShowMobileTokenOnboarding;
+        case 'mobileTokenWithoutTravelcard':
+          return customShouldShowMobileTokenWithoutTravelcardOnboarding;
+        default:
+          return true;
+      }
+    },
+    [
+      customShouldShowExtendedOnboarding,
+      customShouldShowLocationOnboarding,
+      customShouldShowShareTravelHabitsScreen,
+      customShouldShowNotificationPermissionScreen,
+      customShouldShowMobileTokenOnboarding,
+      customShouldShowMobileTokenWithoutTravelcardOnboarding,
+    ],
+  );
+};
+
+/* --- useCustomShouldShow hooks --- */
```

Comment:
What does the term custom mean here actually? I don't understand at first glance.

Reply:
To determine whether an onboarding section should be shown, there is always a check on whether it is already onboarded, which is locally stored with AppContext.
In addition, for each onboarding section, there is also the option to add _custom conditions_ for whether it should be shown. E.g. for location permission onboarding, it should not show if location permission is already granted, and for travel token onboarding, it should only show if the token is active on another device..